### DISCUSSION TITLE
librewolf: Error: no suitable librewolf executable found (flatpak)

### DISCUSSION THREAD

**Discussion by librewolffy** [netblue30/firejail#5113] [Q&A] [answered]
1. Created file librewolf.profile in the directory /etc/firejail using this link: https://github.com/netblue30/firejail/blob/master/etc/profile-a-l/librewolf.profile

2. Commented out the following lines because of errors:
#dbus-user filter
#dbus-user.own io.gitlab.librewolf.*
#dbus-user.own org.mozilla.librewolf.*

3. Enabled the network with "network yes" in /etc/firejail/firejail.config

4. Disabled "restricted-network no"

5. Launched firejail librewolf and got "Error: no suitable librewolf executable found"

Would you advise on the solution for this?

**Comment by ghost** [comment]
> 1. Created file librewolf.profile in the directory /etc/firejail...

If you didn't have that file on your system it seems you're running an outdated version of firejail. In such cases it is not recommended to manually download/copy files from git, as there might be other changes in the firejail binary and/or options in included files that will break things rather easily. Can you provide details on your OS and firejail version?

**Reply by librewolffy** [reply]
I see the Firejail current release is 0.9.68 -> https://firejail.wordpress.com/download-2/

Does this mean I would have to upgrade accordingly from here https://firejail.wordpress.com/download-2/? Will the upgrade include the profile for Librewolf?

**Reply by ghost** [reply]
> The OS is Linux Mint Cinnamon 20.2 and the Firejail version is 0.9.62

Thanks for the info. The best thing you can do on Linux Mint is to follow [these](https://github.com/netblue30/firejail#ubuntu) instructions and add the PPA. It is maintained by one of our collaborators and considered the safest path to keep your firejail packages up to date on your OS.

> Will the upgrade include the profile for Librewolf?

Yes. Firejail 0.9.68 has full support for Librewolf.

**Reply by librewolffy** [reply]
It is shown Firejail 0.9.6.8 but the same error - Error: no suitable librewolf executable found

The part where it says:
Configuration file /etc/firejail/firejail.config'
==> Modified (by....
==> Package distributor...
   What would you like to do...
     Y or I   : install the package maintainer's version

I entered 'y'

Would you advise? Thanks

**Reply by rusty-snake** [reply]
How/Where did you installed librewolf?

**Reply by librewolffy** [reply]
I used Flatpak to install the first time, but it was 9.62

I have just upgraded using the suggested instructions from here: https://github.com/netblue30/firejail#ubuntu
sudo add-apt-repository ppa:deki/firejail
sudo apt-get update
sudo apt-get install firejail firejail-profiles

After that, I was able to see the librewolf.profile listed in /etc/firejail

**Comment by rusty-snake** [comment, marked as answer]
> I used Flatpak to install the first time

Duplicated of #4138. You can not firejail flatpaks, they come with their own sandbox.

**Reply by rusty-snake** [reply]
> 5. Launched firejail librewolf and got "Error: no suitable librewolf executable found"
> Would you advise on the solution for this?

Make sure the `librewolf` command works w/o firejail.

**Reply by librewolffy** [reply]
I didn't download the Flatpak but used the command:
flatpak install flathub io.gitlab.librewolf-community  

$ librewolf
librewolf: command not found

I don't see Librewolf directory on /etc

So would I have to uninstall Librewolf and install the appimage so I can use Firejail with it?

How can this be resolved because I want to use "firejail librewolf" from the command instead.

**Reply by ghost** [reply]
> So would I have to uninstall Librewolf and install the appimage so I can use Firejail with it?
How can this be resolved because I want to use "firejail librewolf" from the command instead.

Yes, uninstall the flatpak version. IMO the easiest, secure and low-maintenance way to get Librewolf + regular updates is to `use your package manager`. Follow the instructions on adding [Librewolf's PPA](https://librewolf.net/installation/debian/). If you use `firecfg`, re-run it after installing the app and from that moment on you can run a firejailed Librewolf just by running `librewolf`.

**Reply by librewolffy** [reply]
Thanks. Awesome. It works. 

How can I mark this as "Answered"?

**Reply by ghost** [reply]
Good news :+1:.

> How can I mark this as "Answered"?

Use one of the `Mark as answer` buttons below the replies.
